extension/src/plots/model/collect.ts

```diff
@@ -312,60 +342,131 @@ export const collectData = (output: PlotsOutput): DataAccumulator => {
   return acc
 }
 
-type ComparisonPlotsAcc = { path: string; revisions: ComparisonRevisionData }[]
+type ComparisonPlotsAcc = {
+  classDetails: ComparisonClassDetails
+  path: string
+  revisions: ComparisonRevisionData
+}[]
 
 type GetComparisonPlotImg = (
   img: ImagePlot,
   id: string,
   path: string
 ) => ComparisonPlotImg
 
+export const boundingBoxColors = [
+  '#ff3838',
+  '#ff9d97',
+  '#ff701f',
+  '#ffb21d',
+  '#cfd231',
+  '#48f90a',
+  '#92cc17',
+  '#3ddb86',
+  '#1a9334',
+  '#00d4bb',
+  '#2c99a8',
+  '#00c2ff',
+  '#344593',
+  '#6473ff',
+  '#0018ec',
+  '#8438ff',
+  '#520085',
+  '#cb38ff',
+  '#ff95c8',
+  '#ff37c7'
+]
+
+const collectComparisonPlotImgs = (
+  boundingBoxClassLabels: Set<string>,
+  imgs: ImagePlot[],
+  getComparisonPlotImg: GetComparisonPlotImg,
+  id: string,
+  path: string
+) => {
+  const plotImgs = []
+  for (const img of imgs) {
+    plotImgs.push(getComparisonPlotImg(img, id, path))
+
+    if (img.boundingBoxes) {
+      for (const label of Object.keys(img.boundingBoxes)) {
+        boundingBoxClassLabels.add(label)
+      }
+    }
+  }
+
+  return plotImgs
+}
+
 const collectSelectedPathComparisonPlots = ({
   acc,
   comparisonData,
+  comparisonClassesSelected,
   path,
   selectedRevisionIds,
   getComparisonPlotImg
 }: {
   acc: ComparisonPlotsAcc
   comparisonData: ComparisonData
+  comparisonClassesSelected: ComparisonClassesSelected
   path: string
   selectedRevisionIds: string[]
   getComparisonPlotImg: GetComparisonPlotImg
 }) => {
+  const boundingBoxClassLabels = new Set<string>()
   const pathRevisions = {
+    classDetails: {} as ComparisonClassDetails,
     path,
     revisions: {} as ComparisonRevisionData
   }
 
   for (const id of selectedRevisionIds) {
-    const imgs = comparisonData[id]?.[path]
+    const imgs: ImagePlot[] | undefined = comparisonData[id]?.[path]
+
     pathRevisions.revisions[id] = {
       id,
       imgs: imgs
-        ? imgs.map(img => getComparisonPlotImg(img, id, path))
+        ? collectComparisonPlotImgs(
+            boundingBoxClassLabels,
+            imgs,
+            getComparisonPlotImg,
+            id,
+            path
+          )
         : [{ errors: undefined, loading: false, url: undefined }]
     }
   }
+
+  for (const [ind, label] of [...boundingBoxClassLabels].entries()) {
```

Comment:
`.entries()` should work directly on `Set`  without conversion.

Reply:
Great idea! Updated to use a variable for getting the index.

Reply:
That's strange, I could have sworn I updated this. Must have reverted it accidentally when I was adding a stash in one of the last commits I added. Apologies, I'll update this!

Reply:
> That's strange, I could have sworn I updated this. Must have reverted it accidentally when I was adding a stash in one of the last commits I added. Apologies, I'll update this!

Updated!